pkg/didmethod/peer/resolver.go

```diff
@@ -21,7 +22,7 @@ func NewDIDResolver(store *DIDStore) *DIDResolver {
 }
 
 // Read implements didresolver.DidMethod.Read interface (https://w3c-ccg.github.io/did-resolution/#resolving-input)
-func (resl *DIDResolver) Read(did string, versionID interface{}, versionTime string, noCache bool) ([]byte, error) {
+func (resl *DIDResolver) Read(did string, opts ...didresolver.Options) ([]byte, error) {
```

Comment:
I like collapsing the optional arguments. As I mention later, I am wondering why use an Option struct instead of functional options?

FYI: #65 also mentions a boolean return value.

Reply:
> FYI: #65 also mentions a boolean return value.

Added - I had overlooked that completely